src/main/java/kr/motd/maven/os/FileOperationProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Trustin Heuiseung Lee.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kr.motd.maven.os;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Interface exposing file operations.
+ */
+public interface FileOperationProvider {
+
+  /**
+   * Gets a {@link InputStream} for reading the content of the file with the specified name.
+   *
+   * @param      fileName   the system-dependent file name.
+   * @return     the {@link InputStream} that can be read to get the file content.
+   * @throws     IOException if the file does not exist, is a directory rather than a regular
+   *             file, or for some other reason cannot be opened for reading.
+   */
+  InputStream readFile(String fileName) throws IOException;
```

Comment:
How about simplifying this into:

```
List<String> readLines(String filePath, Charset charset) throws IOException
```

.. so that the caller doesn't really need to handle buffering, etc?

Reply:
That just pushes the burden to the implementers. Returning an `InputStream` is more general-purpose for implementations. In some cases, implementations are not able to work with the given `Charset` due to API limitations provided by the build tool. For example, Gradle's configuration-phase-safe API for reading files only gives [`getAsBytes()`](https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileContents.html#getAsBytes--) and [`getAsText()`](https://docs.gradle.org/current/javadoc/org/gradle/api/file/FileContents.html#getAsText--). It has no way to respect the given Charset and it would be cumbersome to break a `String` of file back to lines.

Reply:
OK. Then could you rename `fileName` to `filePath` or `path` at least?

Reply:
Absolutely.